### ISSUE TITLE
 @auth(rules: [{ allow: owner, operations: [create, read, update, delete] }] Not working. Still getting: Not Authorized to access deletePost on type Mutation

### ISSUE THREAD

**Issue by BernhardSmuts** [aws-amplify/amplify-cli#9976]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v16.13.0

### Amplify CLI Version

7.6.23

### What operating system are you using?

Windows 10

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No changes were made

### Amplify Categories

auth, storage, function, api, hosting

### Amplify Commands

publish, push

### Describe the bug

@josefaidt  I am still sitting with this same issue. 

**My flag is on**

`"addmissingownerfields": true,`

**this is my Schema**

```type Ship
  @model
  @auth(
    rules: [
      { allow: public, operations: [read] },
      { allow: private, operations: [create, read, update] },
      { allow: owner, operations: [create, read, update, delete] }
    ]
  ) {
  id: ID!
  ShipStatus: String! @index(name: "byActiveShips", sortKeyFields: ["updatedAt"], queryField: "shipByStatus")
  ShipHeading: String!
  ShipDescription: String!
  ShipCategory: String!
  ShipAuthor: String!
  ShipImageStorageKey: String!
  ShipHashtags: String!
  createdAt: AWSDateTime!
  updatedAt: AWSDateTime!
  BoatID: ID
  BoatsCount: String @function(name: "randomfunctionhbe3de0e-${env}")
  Boats: [Boat] @hasMany(indexName: "byShip", fields: ["id"])
}

type Boat
  @model
  @auth(
    rules: [
      { allow: public, operations: [read] },
      { allow: private, operations: [create, read, update] },
      { allow: owner, operations: [create, read, update, delete] }
    ]
  ) {
  id: ID!
  ShipID: ID! @index(name: "byShip", sortKeyFields: ["updatedAt"])
  BoatVersion1MasterID: ID! @index(name: "byVersions", sortKeyFields: ["updatedAt"])
  BoatStatus: String!
  BoatAuthor: String!
  BoatDescription: String!
  BoatImageStorageKey: String
  BoatLikes: Int
  BoatDislikes: Int
  createdAt: AWSDateTime!
  updatedAt: AWSDateTime!
  BoatVersions: [Boat] @hasMany(indexName: "byVersions", fields: ["id"])
}
```

After wiping the DB and adding fields again no owner field is automatically created. 

I even took the suggestion of adding the `owner` field manually and retrying, but no owner field gets created. Do i have to manually insert data into the owner field? What do I put there?
If that's the case they what's the use? I could just write that logic manually?

**Using version 7.6.23**
 
![image](https://user-images.githubusercontent.com/22454448/158186237-22fe6c8f-6bc0-43ee-b4f7-fd8f3d5124b4.png)

This is what the current docs say and it does not act accordingly.

Also tried `{ allow: owner, ownerField: "owner", operations: [create, read, update, delete] }`

Still can't do a simple delete 🤕 

### Expected behavior

The owner should be able to delete an entry they have made.

### Reproduction steps

1. Create the API using the schema
2. Create a "Ship" as a Cognito signed-in user
3. Delete a "Ship" as the same Cognito signed-in user

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line
type Ship
  @model
  @auth(
    rules: [
      { allow: public, operations: [read] },
      { allow: private, operations: [create, read, update] },
      { allow: owner, operations: [create, read, update, delete] }
    ]
  ) {
  id: ID!
  ShipStatus: String! @index(name: "byActiveShips", sortKeyFields: ["updatedAt"], queryField: "shipByStatus")
  ShipHeading: String!
  ShipDescription: String!
  ShipCategory: String!
  ShipAuthor: String!
  ShipImageStorageKey: String!
  ShipHashtags: String!
  createdAt: AWSDateTime!
  updatedAt: AWSDateTime!
  BoatID: ID
  BoatsCount: String @function(name: "randomfunctionhbe3de0e-${env}")
  Boats: [Boat] @hasMany(indexName: "byShip", fields: ["id"])
}

type Boat
  @model
  @auth(
    rules: [
      { allow: public, operations: [read] },
      { allow: private, operations: [create, read, update] },
      { allow: owner, operations: [create, read, update, delete] }
    ]
  ) {
  id: ID!
  ShipID: ID! @index(name: "byShip", sortKeyFields: ["updatedAt"])
  BoatVersion1MasterID: ID! @index(name: "byVersions", sortKeyFields: ["updatedAt"])
  BoatStatus: String!
  BoatAuthor: String!
  BoatDescription: String!
  BoatImageStorageKey: String
  BoatLikes: Int
  BoatDislikes: Int
  createdAt: AWSDateTime!
  updatedAt: AWSDateTime!
  BoatVersions: [Boat] @hasMany(indexName: "byVersions", fields: ["id"])
}

```

</details>

### Log output

<details>

```
# Put your logs below this line
{"data":{"deleteShip":null},"errors":[{"path":["deleteShip"],"data":null,"errorType":"Unauthorized","errorInfo":null,"locations":[{"line":2,"column":3,"sourceName":null}],"message":"Not Authorized to access deleteShip on type Mutation"}]}

```

</details>

### Additional information

Amplify is doing nothing related to the owner field. Not creating it, not writing owner info to it, not pulling it from the DB, not validating against it to allow the delete...

**Comment by BernhardSmuts** [comment]
I managed to find a workaround:

I set these auth rules:
`@auth(rules: [{ allow: owner }, { allow: public, operations: [read] },{ allow: private, operations: [create, read, update] }])`

I set _format on save_ for the schema flag `false` in my editor,
It was removing the comma separator, have no idea if that matters..

Then I manually added the field `owner: String` to both Models.

Then I manually added the sub (user's cognito id) from 
`let userData = await Auth.currentAuthenticatedUser();`
`userData.sub`

I added this to my `create` statment in when writing data to the database and as well and adding the owner field to the get data...

So basically I manually strung the owner field inbound and outbound and now the permissions seem to work as advertised. 

Maybe this info will help someone build the solution in

**Comment by danielleadams** [comment]
Hi @BernhardSmuts, what version of the transformer are you using? You can find it in the `cli.json`. Thank youu.

**Comment by BernhardSmuts** [comment]
I am using the new V2 transformer

**Comment by danielleadams** [comment]
@BernhardSmuts - how are you making the API calls and signing the requests? Would be possible to test this out in AppSync to try to isolate the issue? Thank you.

**Comment by BernhardSmuts** [comment]
I just reconfigured the api as follows:
(removed the manually created "owner:String" field)

type Ship
  @model
  @auth(
    rules: [
      { allow: owner }
      { allow: public, operations: [read] }
      { allow: private, operations: [create, read, update] }
    ]
  ) {
  id: ID!
  shipStatus: String! @index(name: "byActiveShips", sortKeyFields: ["updatedAt"], queryField: "shipByStatus")
  shipHeading: String!
  shipDescription: String!
  shipCategory: String!
  shipAuthor: String!
  shipImageStorageKey: String!
  shipHashtags: String!
  createdAt: AWSDateTime!
  updatedAt: AWSDateTime!
  boatID: ID
  boatsCount: String @function(name: "myappfunctionebe3de0e-${env}")
  boat: [Boat] @hasMany(indexName: "byShip", fields: ["id"])
}

type Boat
  @model
  @auth(
    rules: [
      { allow: owner }
      { allow: public, operations: [read] }
      { allow: private, operations: [create, read, update] }
    ]
  ) {
  id: ID!
  shipID: ID! @index(name: "byShip", sortKeyFields: ["updatedAt"])
  boatVersion1MasterID: ID! @index(name: "byVersions", sortKeyFields: ["updatedAt"])
  boatStatus: String!
  boatAuthor: String!
  boatDescription: String!
  boatImageStorageKey: String
  boatLikes: Int
  boatDislikes: Int
  boatVersions: [Boat] @hasMany(indexName: "byVersions", fields: ["id"])
}

Then I open the appSync console, login with a userpool user and create a Ship.
The owner field is not auto-created and when manually added to the query's  items to return on a `getShip` query, it returns `owner: null`
All the backend was created using the amplify cli, nothing was manually created in the various services' consoles

**Comment by danielleadams** [comment]
Thanks for the clarification @BernhardSmuts. The reason this happens is because the authorization stops once it hits the first rule - in this case it is the `allow: private` rule for User Pools because there are 2 create operations.

Would you be able to assign the `owner` field in the GQL request if you need to keep using `allow: private` rule? Since both rules stem from the same authorization method, could you use just `allow: private` for all the operations and manually assign `owner`? One caveat, though, is that with `owner`, the resolver will verify that the user being authorized is the same as the owner being set, and `private` does not have the same validation.

**Comment by danielleadams** [comment]
Noting that this is a duplicate of https://github.com/aws-amplify/amplify-cli/issues/9920

**Comment by BernhardSmuts** [comment]
Okay thanks, I will look into this a bit further should I run into the issue again.
My solution of manually coding in the owner fields seems to work for now.
Thanks, I'll close this issue.
